### PR TITLE
feature/test-sign-up

### PR DESCRIPTION
**Pull request by wgermancutraro**
## Summary

I made all the test describe in the trello card. I created a folder call "controllers" inside the test folder. There i made a "users" folder and inside i created the users.signup.spec.js file. I like to name the file like that because usually, a controller, has a lot of endpoints, and i found that difficult to manage in one file (could be usersController.spec.js). 

## Known Issues

This is a branch that was born of the sign up branch, thats why is a lot of code under. At first i created a new migration to the table name. I run that and it works. But then when i try to run the server again the error was "migrations pending. npm run migrations". And when i run again that command it says "users dont have a name column'. Thats why i removed that.

## Trello Card

https://trello.com/c/Gu7DVjso/22-tests-de-sign-up

### CODE REVIEW

File: migrations/migrations/20210519174318-users-table.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ module.exports = {
         primaryKey: true,
         type: DataTypes.INTEGER
       },
-      name: {
+      first_name: {
```
Comment (martinremis) [inline migrations/migrations/20210519174318-users-table.js]: You cannot edit migrations. You have to create a new one changing the affected tables and/or columns.

File: test/controllers/users/users.signup.spec.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      }
+    });
+
+    expect(statusCode).toEqual(201);
+    expect(body.token).toBeDefined();
+    expect(body.user).toBeDefined();
+    expect(createdUser.firstName).toEqual('John');
+    expect(createdUser.lastName).toEqual('Doe');
+    expect(createdUser.email).toEqual('jdoe@wolox.com.ar');
+  });
```
Comment (martinremis) [inline test/controllers/users/users.signup.spec.js:28]: Not as correction, but as additional information:
Sometimes we prefer to use an structure like the following, to get a better failing tracking (every expect has it own track)
```js
describe('some tests', () => {
  let response;
  let createdUser;

  beforeAll(async () => {
    response = await request(app)
      .post('/users/signup')
      .send({ firstName: 'John', lastName: 'Doe', email: 'jdoe@wolox.com.ar', password: 'Sherman33' });
    createdUser = await UserModel.findOne({ where: { email: 'jdoe@wolox.com.ar' } });
  });

  it('should response 201', () => {
    expect(statusCode).toEqual(201);
  });
  it('should have property token', () => {
    expect(body.token).toBeDefined();
  });
  it('should have property user', () => {
    expect(body.user).toBeDefined();
  });
});
```

**Review by martinremis — CHANGES_REQUESTED** [review]

File: migrations/migrations/20210519174318-users-table.js (hunk 1 above)
Comment (wgermancutraro) [reply migrations/migrations/20210519174318-users-table.js]: i created a new migration, then i removed the user schema of my local postgreSQL (weird error) and it works (with npm migrations)

Comment (martinremis) [reply migrations/migrations/20210519174318-users-table.js]: I think that is the expected problem when you modify a migration :laughing:. May not be exactly this one, but it creates inconsistencies.

**Review by martinremis — APPROVED** [review]

**Pull request merged**